compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -292,16 +292,16 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
       if (!sym.exists || (sym eq defn.LanguageModuleClass)) ""
       else toPrefix(sym.owner) + sym.name + "."
     def featureName = toPrefix(owner) + feature
-    def hasImport(implicit ctx: Context): Boolean = {
-      if (ctx.importInfo == null || (ctx.importInfo.site.widen.typeSymbol ne owner)) false
-      else if (ctx.importInfo.excluded.contains(feature)) false
```

Comment:
Previously, we stopped and returned false for an unimported feature, but we now continue looking up the chain for an import, this breaks the last three errors in the neg test https://github.com/lampepfl/dotty/blob/master/tests/neg/dynamicNoImport.scala

Reply:
Yes, indeed, the logic is quite subtle here (and it was not documented either).